dev/run-tests.py

```diff
@@ -1,4 +1,4 @@
-#!/usr/bin/env python2
+#!/usr/bin/env python
```

Comment:
I know the intention, but this is worse historically. I'm -1 for this change.
If we are ready, `python3` might be a better choice.

Reply:
If so, should we also change this line in `dev/run-test` ?
```
exec python -u ./dev/run-tests.py "$@"
```

Reply:
For this one, please make a decision first like https://github.com/apache/spark/pull/25289#pullrequestreview-271749485.